mitodl/ocw-hugo-projects
# Update Resource and External Resource Config in www

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4245

### Description (What does it do?)
1. Updates `Resource` configuration to match `course-v2`. This includes `image_metadata`, metadata for `video`, `video_files`, `license`,`learning_resource_type`, and `file_type`.

2. Adds `External Resource`. Allows it to be linked in `pages`, `course-lists`, and `course-collections`.

This PR should be tested along https://github.com/mitodl/ocw-hugo-themes/pull/1412
This is a related ticket for UI/UX changes: https://github.com/mitodl/hq/issues/4260

### Steps to test
To test in ocw-studio, follow the test steps mentioned here with `ocw-hugo-themes` branch name `ibrahim/4245/external-resource-rendering-www` https://github.com/mitodl/ocw-hugo-projects/pull/294.

Also, an easier way might be to add this branch's changes (`ocw-hugo-projects` updated file) in `ocw-studio-rc` for `ocw-www` WebsiteStarter, and then you can add the relevant content in `ocw-www` in Studio RC and publish it.

And then in your local `ocw-hugo-projects`, checkout to this branch. And in `ocw-hugo-themes`, use the `ibrahim/4245/external-resource-rendering-www` branch, get a fresh copy of ocw-www, and run it. `yarn start www`. Make sure you revert the config changes in `ocw-studio-rc` after you're done.

Test the changes in this PR are working as expected. Test Resource for image & video in `ocw-www`.
Test External Resource, make sure it is being linked in `ocw-www` theme correctly. Make sure the warning pop-up appears for non-OCW external resources.
